src/fileio.cpp

```diff
@@ -481,6 +482,25 @@ FILE *FioFOpenFile(const char *filename, const char *mode, Subdirectory subdir,
 		strecpy(resolved_name, filename, lastof(resolved_name));
 		strtolower(resolved_name);
 
+		/* Resolve ".." */
+		std::stringstream ss(resolved_name);
```

Comment:
streams are a bit ick. Is there no existing "string split" function that could be used? If nothing else `std::istringstream` might be a bit better.